fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSTransaction.java

```diff
@@ -0,0 +1,1299 @@
+            List<String> partitionValues,
+            boolean deleteData) {
+        DatabaseTableName databaseTableName = new DatabaseTableName(databaseName, tableName);
+        Map<List<String>, Action<PartitionAndMore>> partitionActionsForTable =
+                partitionActions.computeIfAbsent(databaseTableName, k -> new HashMap<>());
+        Action<PartitionAndMore> oldPartitionAction = partitionActionsForTable.get(partitionValues);
+        if (oldPartitionAction == null) {
+            if (deleteData) {
+                partitionActionsForTable.put(partitionValues, new Action<>(ActionType.DROP, null));
+            } else {
+                partitionActionsForTable.put(partitionValues, new Action<>(ActionType.DROP_PRESERVE_DATA, null));
+            }
+            return;
+        }
+        switch (oldPartitionAction.getType()) {
+            case DROP:
+            case DROP_PRESERVE_DATA:
+                throw new RuntimeException(
+                        "Not found partition from partition actions for " + databaseTableName
+                                + ", partitions: " + partitionValues);
+            case ADD:
+            case ALTER:
+            case INSERT_EXISTING:
+            case MERGE:
+                throw new RuntimeException(
+                        "Dropping a partition added in the same transaction is not supported: "
+                                + databaseTableName + ", partition values: " + partitionValues);
+            default:
+                throw new IllegalStateException("Unknown action type: " + oldPartitionAction.getType());
+        }
+    }
+
+    class HmsCommitter {
+
+        // update statistics for unPartitioned table or existed partition
+        private final List<UpdateStatisticsTask> updateStatisticsTasks = new ArrayList<>();
+        Executor updateStatisticsExecutor = Executors.newFixedThreadPool(16);
+
+        // add new partition
+        private final AddPartitionsTask addPartitionsTask = new AddPartitionsTask();
+
+        // for file system rename operation
+        // whether to cancel the file system tasks
+        private final AtomicBoolean fileSystemTaskCancelled = new AtomicBoolean(false);
+        // file system tasks that are executed asynchronously, including rename_file, rename_dir
+        private final List<CompletableFuture<?>> asyncFileSystemTaskFutures = new ArrayList<>();
+        // when aborted, we need to delete all files under this path, even the current directory
+        private final Queue<DirectoryCleanUpTask> directoryCleanUpTasksForAbort = new ConcurrentLinkedQueue<>();
+        // when aborted, we need restore directory
+        private final List<RenameDirectoryTask> renameDirectoryTasksForAbort = new ArrayList<>();
+        // when finished, we need clear some directories
+        private final List<String> clearDirsForFinish = new ArrayList<>();
+        Executor fileSystemExecutor = Executors.newFixedThreadPool(16);
```

Comment:
I think this should be shared by all transactions? Not for single transaction.

Reply:
This mainly saves some tasks running in the current transaction. These tasks do not isolate the transaction unless there is another task management.